js/oceanex.js

```diff
@@ -0,0 +1,323 @@
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            let id = market['id'].toUpperCase ();
+            let symbol = market['name'];
+            let pair = market['name'].split ('/');
+            let base = pair[0];
+            let quote = pair[1];
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': base,
+                'quoteId': quote,
+                'active': true,
+                'info': market,
+                'precision': undefined,
+                'limits': undefined,
+            });
+        }
+        // parse markets
+        return result;
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        let response = await this.publicGetTickersPair (this.extend ({ 'pair': symbol }), params);
+        const data = response['data'];
+        return this.parseTicker (data, symbol);
+    }
+
+    parseTicker (data, market = undefined) {
+        let timestamp = data['at'];
+        const ticker = data['ticker'];
+        if (typeof timestamp === 'string') {
+            if (timestamp.length > 0) {
+                timestamp = this.parse8601 (timestamp);
+            }
+        }
+        return {
+            'symbol': market,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeFloat (ticker, 'high'),
+            'low': this.safeFloat (ticker, 'low'),
+            'bid': this.safeFloat (ticker, 'buy'),
+            'bidVolume': undefined,
+            'ask': this.safeFloat (ticker, 'sell'),
+            'vwap': undefined,
+            'open': undefined,
+            'close': this.safeFloat (ticker, 'last'),
+            'last': this.safeFloat (ticker, 'last'),
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeFloat (ticker, 'volume'),
+            'quoteVolume': undefined,
+            'info': ticker,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        let response = await this.publicGetOrderBook (this.extend ({
+            'market': symbol,
+            'limit': limit,
+        }, params));
+        const orderbook = this.safeValue (response, 'data');
+        return this.parseOrderBook (orderbook, orderbook['timestamp']);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        let response = await this.publicGetTrades (this.extend ({
+            'market': symbol,
+            'limit': limit,
+        }, params));
+        let data = this.safeValue (response, 'data');
+        return this.parseTrades (data, symbol, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        let side = undefined;
+        if (trade['side'] === 'bid') {
```

Comment:
We need to make sure that we access the keys safely everywhere in all parsers, as described in the CONTRIBUTING guidelines here: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sanitizing-input-with-safe-methods